routes/alfresco.py

```diff
@@ -0,0 +1,478 @@
+
+    Args:
+        var_ep (str): variable endpoint. Either flammability or veg_change
+        lat (float): latitude
+        lon (float): longitude
+        
+    Returns:
+        JSON-like dict of relative flammability data
+        from IEM ALFRESCO outputs
+
+    Notes:
+        example request: http://localhost:5000/flammability/point/65.0628/-146.1627
+    """
+    validation = validate_latlon(lat, lon)
+    if validation == 400:
+        return render_template("400/bad_request.html"), 400
+    if validation == 422:
+        return (
+            render_template(
+                "422/invalid_latlon.html", west_bbox=WEST_BBOX, east_bbox=EAST_BBOX
+            ),
+            422,
+        )
+
+    x, y = project_latlon(lat, lon, 3338)
+
+    if var_ep in var_ep_lu.keys():
+        cov_id_str = var_ep_lu[var_ep]["cov_id_str"]
+        try:
+            point_data_list = asyncio.run(fetch_alf_point_data(x, y, cov_id_str))
+        except Exception as exc:
+            if hasattr(exc, "status") and exc.status == 404:
+                return render_template("404/no_data.html"), 404
+            return render_template("500/server_error.html"), 500
+    else:
+        return render_template("400/bad_request.html"), 400
+
+    varname = var_ep_lu[var_ep]["varname"]
+    point_pkg = package_historical_alf_point_data(point_data_list[0], varname)
+    # package AR5 data and fold into data pakage
+    ar5_point_pkg = package_ar5_alf_point_data(point_data_list[3], varname)
+    for era, summaries in ar5_point_pkg.items():
+        point_pkg[era] = summaries
+    # package summary future eras in
+    point_pkg["2040-2069"] = package_ar5_alf_averaged_point_data(
+        point_data_list[1], varname
+    )
+    point_pkg["2070-2099"] = package_ar5_alf_averaged_point_data(
+        point_data_list[2], varname
+    )
+
+    if request.args.get("format") == "csv":
+        csv_data = create_csv(point_pkg, alf_fieldnames, varname=varname, stat="mean")
+        return return_csv(csv_data)
+
+    return postprocess(point_pkg, "alfresco")
+
+
+@routes.route("/alfresco/<var_ep>/huc/<huc_id>")
+def run_fetch_alf_huc_data(var_ep, huc_id):
+    """HUC-aggregation data endpoint. Fetch data within HUC
+    for specified variable and return JSON-like dict.
+
+    Args:
+        var_ep (str): variable endpoint. Either veg_change or flammability
+        huc_id (int): 8-digit HUC ID
+    Returns:
+        huc_pkg (dict): zonal mean of variable(s) for HUC polygon
+
+    """
+    validation = validate_huc8(huc_id)
+    if validation == 400:
+        return render_template("400/bad_request.html"), 400
+    try:
+        huc_pkg = run_aggregate_var_polygon(var_ep, huc8_gdf, huc_id)
+    except:
+        return render_template("422/invalid_huc.html"), 422
+
+    if request.args.get("format") == "csv":
+        varname = var_ep_lu[var_ep]["varname"]
+        csv_data = create_csv(huc_pkg, alf_fieldnames, varname=varname, stat="mean")
```

Comment:
Here and in the point & protected area, the columns headers don't match the data (opened in Numbers on MacOS), but the CSV agrees.

<img width="448" alt="Screen Shot 2022-02-15 at 11 09 25 AM" src="https://user-images.githubusercontent.com/525049/154141309-8884dbaa-0ad2-4bb5-a005-2056c3181af9.png">

Reply:
fixed